assets/js/googlesitekit/datastore/user/key-metrics.js

```diff
@@ -305,6 +332,18 @@ const baseSelectors = {
 	getKeyMetricsSettings( state ) {
 		return state.keyMetricsSettings;
 	},
+
+	/**
+	 * Determines whether the key metrics settings are being saved or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} TRUE if the key metrics settings are being saved, otherwise FALSE.
+	 */
+	isSavingKeyMetricsSettings( state ) {
+		return !! state?.isSavingKeyMetricsSettings;
+	},
```

Comment:
We should be able to do this without the additional saving state. We've needed to do the same a few times before. Here's one example https://github.com/google/site-kit-wp/blob/2dffea28d9e449cab27159189f59e77285ba3a93/assets/js/modules/analytics/datastore/profiles.js#L257-L261